packages/libs/error-stack/src/fmt/color.rs

```diff
@@ -0,0 +1,153 @@
+#[cfg(feature = "color")]
+use crate::fmt::r#override::{AtomicOverride, AtomicPreference};
+use crate::Report;
+
+/// The available modes of color support
+///
+/// Can be accessed through [`crate::fmt::HookContext::color_mode`], and set via
+/// [`Report::set_color_mode`]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ColorMode {
+    /// User preference to disable all colors
+    // TODO: this is only true once https://github.com/jam1garner/owo-colors/pull/90 is merged
+    // If this is the variant is present, [`owo-colors`](https://docs.rs/owo-colors) color
+    // support has been temporarily disabled and closures given to
+    // [`OwoColorize::if_supports_color`] will not be executed.
+    None,
+
+    /// User preference to enable colors
+    // TODO: this is only true once https://github.com/jam1garner/owo-colors/pull/90 is merged
+    // This will also temporarily set [`owo_colors::set_override`] to enable execution of
+    // [`OwoColorize::if_supports_color`].
+    Color,
+
+    /// User preference to enable styles, but discourage colors
+    ///
+    /// This is the same as [`ColorMode::Color`], but signals to the user that while colors are
+    /// supported, the user prefers instead the use of emphasis, like bold and italic text.
+    Emphasis,
+}
+
+impl ColorMode {
+    #[cfg(feature = "color")]
+    pub(super) fn load() -> Self {
+        // The default is `ColorMode::Emphasis`, as colors are hard. ANSI colors are not
+        // standardized, and some colors may not show at all.
+        COLOR_OVERRIDE.load().map_or(Self::Emphasis, |mode| mode)
+    }
+
+    #[cfg(not(feature = "color"))]
+    pub(super) const fn load() -> Self {
+        ColorMode::None
+    }
+}
+
+/// Value layout:
+/// `0x00`: `ColorMode::None`
+/// `0x01`: `ColorMode::Color`
+/// `0x02`: `ColorMode::Emphasis`
+///
+/// all others: unset/none
+#[cfg(feature = "color")]
+impl AtomicPreference for ColorMode {
+    fn load(value: u8) -> Option<Self> {
```

Comment:
Why do we have two implementations of `load` with different semantics?

Reply:
fixed in [f83228f](https://github.com/hashintel/hash/pull/1800/commits/f83228f7d0c91828e09c642bd916119794d17402)